### PR TITLE
mbta elevator hotline

### PR DESCRIPTION
**Pull request by mogorman**
Task: [Pull alerts from V3 API](https://app.asana.com/0/1158387316054713/1177701938360845)

This is the initial work for the new mbta elevator hotline.

It is implemented via aws connect, a simple lambda, and the MBTA apiv3.

The aws connect contact flow is attached as hotline_ivr.json and a screenshot as hotline_ivr.png

the lambda that is called is index.js where all the functionality is defined in the src directory.

claudia is currently being used for deploys and local verification.

the MBTA api calls are contained in `src/alerts.js` and `src/stations.js`

the meat of the lambda is in `src/lambda.js` which takes the json receieved and converts it to an ssml json object to be processed by the contact flow.

### CODE REVIEW

File: claudia.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (ryan-mahoney) [inline claudia.json:1]: do we want to include this? Don't think we'll be deploying the app with claudia

File: src/alerts.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+const client = require("./api_client.js");
+const alertsUrl =
+  "https://api-v3.mbta.com/alerts?fields%5Balert%5D=updated_at%2Ceffect%2Cdescription%2Cheader%2Cinformed_entity&filter%5Bactivity%5D=ALL";
```
Comment (ryan-mahoney) [inline src/alerts.js:3]: Can this be an environment variable? (the `https://api-v3.mbta.com` part)

File: src/alerts.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+exports.get = () => {
+  return client.get(alertsUrl).then((response) => {
+    return {
+      id: "alerts",
+      alerts: response.data
+        .filter((alert) => {
+          return (
+            alert.attributes.effect === "ELEVATOR_CLOSURE" ||
+            alert.attributes.effect === "ESCALATOR_CLOSURE"
+            //     || alert.attributes.effect === "ACCESS_ISSUE"
```
Comment (ryan-mahoney) [inline src/alerts.js:18]: we typically don't include commented out code in commits -- I think this is something that could easily be added in later if necessary

File: src/alerts.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          );
+        })
+        .map((alert) => {
+          const a = alert.attributes;
```
Comment (ryan-mahoney) [inline src/alerts.js:22]: `a` could be a bit more descriptive :)

File: src/alerts.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+          const description = [
+            a.effect.toLowerCase().replace("_", " "),
+            emptyStringIfNull(a.header),
+            emptyStringIfNull(a.description),
+          ].join(" ");
+
+          const stations = a.informed_entity.map((station) => {
```
Comment (ryan-mahoney) [inline src/alerts.js:29]: Might be possible to write as one-liner:
```
const stations = a.informed_entity.map(station => station.stop);
```

File: src/api_client.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+const https = require("https");
+const alertsUrl = "https://api-v3.mbta.com/alerts";
```
Comment (ryan-mahoney) [inline src/api_client.js:2]: env var comment again

File: src/stations.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+const client = require("./api_client.js");
+
+exports.get = (line, routes) => {
+  const url =
+    "https://api-v3.mbta.com/stops?fields%5Bstop%2Cname%5D=stop&filter%5Broute%5D=" +
```
Comment (ryan-mahoney) [inline src/stations.js:5]: env var again

File: src/lambda.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
+const alerts = require("./alerts.js");
```
Comment (ryan-mahoney) [inline src/lambda.js:1]: from a testing/readability perspective, might be good to break this file into some smaller functions that can be tested individually

File: test/index.test.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+const lambda = require("../index.js");
+
+test("fake test till I write some", () => {
```
Comment (ryan-mahoney) [inline test/index.test.js:3]: will need some real tests in order to approve

File: claudia.json (hunk 1 above)
Comment (mogorman) [reply claudia.json:1]: from talking to ian we dont. but currently this is my deployment path until someone shows me how to terraform it correctly.

File: src/alerts.js (hunk 3 above)
Comment (mogorman) [reply src/alerts.js:18]: agreed. i was leaving it in as i wasn't sure if we wanted this or not. after talking with katie i think it should be deleted

**Pull request merged**